sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,252 @@
+#title          :discretization.py
+#description    :Implementation of a few discretization algorithms
+#author         :Henry Lin <hlin117@gmail.com>
+#version        :0.0.1
+#python_version :2.7.6
+#==============================================================================
+
+from collections import Counter
+from math import log
+import numpy as np
+from itertools import izip, islice
+
+
+def log2(x):
+    return log(x, 2) if x > 0 else 0
+
+
+class MDLP(object):
+    """Implements the MDLP algorithm from Usama Fayyad's paper
+    "Multi-Interval Discretization of Continuous-Valued Attributes for
+    Classification Learning"
+    """
+
+    def __init__(self):
```

Comment:
Please review the [contributor's guide](http://scikit-learn.org/stable/developers/index.html). `__init__` might take constructor parameters like `continuous_features`, and these -- but only these -- should set attributes of the same names on the object.